src/m365/identity/commands/identity-set.ts

```diff
@@ -0,0 +1,136 @@
+import { Logger } from '../../../cli/Logger.js';
+import auth, { Identity } from '../../../Auth.js';
+import commands from "../commands.js";
+import Command, { CommandError } from '../../../Command.js';
+import GlobalOptions from '../../../GlobalOptions.js';
+import { validation } from '../../../utils/validation.js';
+import { formatting } from '../../../utils/formatting.js';
+import { Cli } from '../../../cli/Cli.js';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  id?: string;
+  name?: string;
+}
+
+class IdentitySetCommand extends Command {
+  public get name(): string {
+    return commands.SET;
+  }
+
+  public get description(): string {
+    return "Switches to another identity, when signed into multiple identities";
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initValidators();
+    this.#initOptions();
+    this.#initOptionSets();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        id: typeof args.options.id !== 'undefined',
+        name: typeof args.options.name !== 'undefined'
+      });
+    });
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.id && !validation.isValidGuid(args.options.id as string)) {
+          return `${args.options.id} is not a valid GUID`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-i, --id [id]'
+      },
+      {
+        option: '-n, --name [name]'
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['id', 'name'] });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    const identity = await this.getIdentityToSwitchTo(args.options);
+
+    if (this.verbose) {
+      await logger.logToStderr(`Switching to identity '${identity.identityName}'...`);
+    }
+
+    await auth.switchToIdentity(identity);
+
+    try {
+
+      if (this.verbose) {
+        logger.logToStderr(`Ensuring identity access token valid...`);
+      }
+
+      await auth.ensureAccessToken(auth.defaultResource, logger, this.debug);
```

Comment:
Is there a reason why we're doing this? Seems like an unnecessary guess as to which commands the user will need. I suggest we remove it and let the token be retrieved when it's needed based on the needed resources.

Reply:
The reason is in some cases the token is expired and cannot be refreshed. I figured you'd want to know immediately...

Reply:
What if refreshing the token fails: do we fail setting the identity? I wonder if we're not trying to cram too many responsibilities here.

Reply:
ok, I can remove it. no problem :-)